lib/flow/VaporField.cpp

```diff
@@ -0,0 +1,424 @@
+#include "vapor/VaporField.h"
+#include <sstream>
+
+using namespace flow;
+
+// Constructor
+VaporField::VaporField( size_t cache_limit )
+    : _recentGrids( cache_limit )
+{ }
+
+
+bool
+VaporField::InsideVolumeVelocity( float time, const glm::vec3& pos )
+{
+    const std::vector<double> coords{ pos.x, pos.y, pos.z };
+    const VAPoR::Grid* grid = nullptr;
+    assert( _isReady() );
+
+    // In case of steady field, we only check a specific time step
+    if( IsSteady )
+    {
+        size_t currentTS = _params->GetCurrentTimestep();
+        for( auto v : VelocityNames )   // cannot use reference here...
+            if( !v.empty() )
+            {
+                int rv = _getAGrid( currentTS, v, &grid );
+                assert( rv == 0 );
+                if( !grid->InsideGrid( coords ) )
+                    return false;
+            }
+    }
+    else    // we check two time steps
+    {
+        // First check if the query time is within range
+        if( time < _timestamps.front() || time > _timestamps.back() )
+            return false;
+
+        // Then locate the 2 time steps
+        size_t floor;
+        int rv  = LocateTimestamp( time, floor );
+        if( rv != 0 ) return false;
+
+        // Second test if pos is inside of time step "floor"
+        for( auto v : VelocityNames )   // cannot use references...
+            if( !v.empty() )
+            {
+                rv = _getAGrid( floor, v, &grid );
+                assert( rv == 0 );
+                if( !grid->InsideGrid( coords ) )
+                    return false;
+            }
+
+        // If time is larger than _timestamps[floor], we also need to test _timestamps[floor+1]
+        if( time > _timestamps[floor] )
+        {
+            for( auto v : VelocityNames )   // cannot use references
+                if( !v.empty() )
+                {
+                    rv = _getAGrid( floor + 1, v, &grid );
+                    assert( rv == 0 );
+                    if( !grid->InsideGrid( coords ) )
+                        return false;
+                }
+        }
+    }
+
+    return true;
+}
+
+
+bool
+VaporField::InsideVolumeScalar( float time, const glm::vec3& pos )
+{
+    if( ScalarName.empty() )
+        return NO_FIELD_YET;
```

Comment:
Won't this result in a true value (the method returns a boolean)? Is that what you want? Why not just return false?

Reply:
Good point, now it just returns false.